demo/site/src/util/createNextImageSizes.ts

```diff
@@ -0,0 +1,31 @@
+type Breakpoints = {
+    sm: number;
+    md: number;
+    lg: number;
+    xl: number;
+};
+
+const breakpointMapping: Record<keyof Breakpoints, number> = {
+    sm: 600,
+    md: 900,
+    lg: 1200,
+    xl: 1600,
+};
+
+export function createNextImageSizes(breakpointWidths: Partial<Record<keyof Breakpoints, string | number>>, defaultWidth: string | number = "100vw") {
```

Comment:
Should we remove the defaultWidth value and instead add a xs breakpoint to breakpointWidths?